frontend/exporter/src/traits.rs

```diff
@@ -200,62 +199,90 @@ pub mod rustc {
                     })
                     .collect()
             }
+        }
 
-            #[tracing::instrument(level = "trace", skip(s))]
-            fn path_to(
-                self,
-                s: &S,
-                target: PolyTraitRef<'tcx>,
-                param_env: rustc_middle::ty::ParamEnv<'tcx>,
-            ) -> Option<Path<'tcx>> {
-                let tcx = s.base().tcx;
-                if predicate_equality(self.upcast(tcx), target.upcast(tcx), param_env, s) {
-                    return Some(vec![]);
+        #[tracing::instrument(level = "trace", skip(s, param_env))]
+        pub fn path_to<'tcx, S: UnderOwnerState<'tcx>>(
+            starting_points: Vec<AnnotatedPredicate<'tcx>>,
```

Comment:
Maybe you can take a `&[AnnotatedPredicate<'tcx>]` instead of a vec here?
Then you would not need to clone on call site.